common/src/main/kotlin/io/homeassistant/companion/android/common/data/integration/impl/entities/UpdateLocationRequest.kt

```diff
@@ -1,14 +1,94 @@
 package io.homeassistant.companion.android.common.data.integration.impl.entities
 
-import com.fasterxml.jackson.annotation.JsonInclude
+import kotlin.reflect.jvm.jvmName
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.builtins.DoubleArraySerializer
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.CompositeDecoder
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.encoding.decodeStructure
+import kotlinx.serialization.encoding.encodeStructure
+import kotlinx.serialization.json.JsonDecoder
+import kotlinx.serialization.json.JsonEncoder
 
-@JsonInclude(JsonInclude.Include.NON_NULL)
+@Serializable(with = UpdateLocationRequestSerializer::class)
 data class UpdateLocationRequest(
-    val gps: Array<Double>?,
-    val gpsAccuracy: Int?,
-    val locationName: String?,
-    val speed: Int?,
-    val altitude: Int?,
-    val course: Int?,
-    val verticalAccuracy: Int?
+    val gps: List<Double>? = null,
+    val gpsAccuracy: Int? = null,
+    val locationName: String? = null,
+    val speed: Int? = null,
+    val altitude: Int? = null,
+    val course: Int? = null,
+    val verticalAccuracy: Int? = null,
 )
+
+/**
+ * Since our global serializer is printing the null value we need to use the internal serializer
+ * to avoid this behavior.
```

Comment:
Describe why we're avoiding this.

```suggestion
 * The global serializer will encode null values by default. For this request, that behavior
 * is unwanted as not all location updates contain all data and the server requires non-null
 * values, so this serializer is used to avoid null values being encoded.
```